### ISSUE TITLE
Bug: `FINAL_PEER` being returned without diallable addresses 

### ISSUE THREAD

**Issue by maschad** [ipfs/js-kubo-rpc-client#176]
I am running a local kubo daemon which I am querying via `peerID` i.e.

```
curl -X POST "http://127.0.0.1:5001/api/v0/dht/findpeer?arg=<my-peer-id>" 

```

and I do note that the `FINAL_PEER` event is returned without a diallable address and no `QUERY_ERROR` but I notice the [previous js-ipfs RPC client implementation](https://github.com/ipfs/js-ipfs/blob/master/packages/interface-ipfs-core/src/dht/find-peer.js#L66-L74) would return a `QUERY_ERROR`  if a peer is not found during a DHT lookup, and so a `FINAL_PEER` event  should [not be returned if it is unable to find the peer in the dht](https://github.com/ipfs/js-kubo-rpc-client/blob/f326ceea41bc0b20723d0ef6fa20844c7fd7b522/test/interface-tests/src/dht/find-peer.js#L65C5-L93) as mentioned in the comment this may be an issue with the Kubo HTTP API.

This was mentioned in https://github.com/ipfs/js-kubo-rpc-client/issues/56 but I am creating a separate issue as this relates directly to https://github.com/libp2p/js-libp2p/issues/1951#issuecomment-1680875313 and https://github.com/libp2p/js-libp2p/issues/1985

**Comment by SgtPooki** [comment]
@maschad we may need to return a QUERY_ERROR from https://github.com/libp2p/js-libp2p-delegated-peer-routing if Kubo is operating as expected, as the kubo-rpc-client library is intended to duplicate direct calls to the HTTP API..

I'll attempt to get confirmation of the expected functionality.

**Comment by SgtPooki** [comment]
Kubo-rpc-client is returning the exact same as Kubo, which means this is not a bug with this library IMO.. I opened https://github.com/ipfs/kubo/issues/10117 and need to wait on further guidance/infi from Kubo folks here.

**Comment by achingbrain** [comment]
FWIW I don't think this is a case for `QUERY_ERROR` - it's for when dialling a node failed or they return an invalid response (e.g. a bad protobuf) or similar.

In this case the query executed without error, there just weren't any results - there should be no `FINAL_PEER` event.

**Comment by SgtPooki** [comment]
Next action items are unclear on this. Could use some guidance from @hacdias or @lidel on how we want to handle this. 

is a code-change in kubo-rpc-client required, or are we just waiting for https://github.com/ipfs/kubo/issues/10117 to release

**Comment by BigLep** [comment]
Per 2023-09-15 verbal, my understanding is this will be handled in Kubo 0.23 release: https://github.com/ipfs/kubo/issues/10014
